ingestion/src/metadata/ingestion/source/pipeline/airflow.py

```diff
@@ -169,14 +169,17 @@ def fetch_pipeline_status(
                 )
                 for task in tasks
             ]
-            pipeline_status = PipelineStatus(
-                taskStatus=task_statuses,
-                executionStatus=STATUS_MAP.get(dag._state, StatusType.Pending.value),
-                executionDate=dag.start_date.timestamp(),
-            )
-            yield OMetaPipelineStatus(
-                pipeline_fqn=pipeline_fqn, pipeline_status=pipeline_status
-            )
+            if dag.start_date:
+                pipeline_status = PipelineStatus(
+                    taskStatus=task_statuses,
+                    executionStatus=STATUS_MAP.get(
+                        dag._state, StatusType.Pending.value
+                    ),
+                    executionDate=dag.start_date.timestamp(),
```

Comment:
Instead of ignoring the entire pipeline status if the date is not available we what we can do is `executionDate=dag.start_date.timestamp() if dag.start_date else None` because the executionDate is an optional field

Reply:
yeah actually IDK why it is not required 🤔  we might need to fix it at some point.
Here we could pick up dag.execution_date, which should always be there and better reflects when it runs

Reply:
nice catch btw